Add CSSFontPaletteValuesRule.* (#25476)
mdn/content: files/en-us/web/api/cssfontpalettevaluesrule/overridecolors/index.md

---
title: CSSFontPaletteValuesRule.overrideColors
slug: Web/API/CSSFontPaletteValuesRule/overrideColors
page-type: web-api-instance-property
browser-compat: api.CSSFontPaletteValuesRule.overrideColors
---

{{APIRef("CSSOM")}}

The read-only **`overrideColors`** property of the {{domxref("CSSFontPaletteValuesRule")}} interface is a string containing a list of color index and color pair that are to be used instead. It is specified in the same format as the corresponding {{cssxref("@font-palette-values/override-colors", "override-colors")}} descriptor.

## Value

A string containing a comma-separated list of color index and color pair

## Examples

> ### Read the overriden colors

Reviewer: ```suggestion
### Reading the overridden colors
```

    format("woff2");
}

.emoji {
  font-family: "Noto Color Emoji";
  font-size: 3rem;
}

@font-palette-values --blue {
  font-family: "Noto Color Emoji";
  override-colors: 3 rgb(1, 28, 193), 4 rgb(60, 124, 230);
}

@font-palette-values --green {
  font-family: "Noto Color Emoji";
  override-colors: 3 rgb(28, 193, 1), 4 rgb(34, 230, 1);
}

.colored-hat {
  font-palette: --blue;
}
```

#### JavaScript

```js
const log = document.getElementById("log");
const button = document.querySelector("button");
const hat = document.querySelector(".colored-hat");
const rules = document.styleSheets[document.styleSheets.length - 1].cssRules;
const greenFontPaletteValuesRule = rules[3];
const blueFontPaletteValuesRule = rules[2];
log.textContent = `Overriden colors: ${blueFontPaletteValuesRule.overrideColors}`;

button.addEventListener("click", (event) => {
  if (hat.style.fontPalette !== "--green") {
    hat.style.fontPalette = "--green";
    log.textContent = `Overriden colors: ${greenFontPaletteValuesRule.overrideColors}`;
  } else {
    hat.style.fontPalette = "--blue";
    log.textContent = `Overriden colors: ${blueFontPaletteValuesRule.overrideColors}`;
  }
});
```

#### Result

> {{EmbedLiveSample("Read the overriden colors", "100", "125")}}

Reviewer: ```suggestion
{{EmbedLiveSample("Reading the overridden colors", "100", "125")}}
```

## Specifications

{{Specifications}}

## Browser compatibility

{{Compat}}

## See also

> - {{cssxref("@font-palette-values")}} at-rule
> - {{cssxref("@font-palette-values/override-colors", "override-colors")}} descriptor

Reviewer: ```suggestion
- {{cssxref("@font-palette-values")}}
- {{cssxref("@font-palette-values/override-colors", "override-colors")}}
```

---
title: CSSFontPaletteValuesRule.overrideColors
slug: Web/API/CSSFontPaletteValuesRule/overrideColors
page-type: web-api-instance-property
browser-compat: api.CSSFontPaletteValuesRule.overrideColors
---

{{APIRef("CSSOM")}}

> The read-only **`overrideColors`** property of the {{domxref("CSSFontPaletteValuesRule")}} interface is a string containing a list of pairs of color indices and the color that are to be used instead, in the same format as the corresponding {{cssxref("@font-palette-values/override-colors", "override-colors")}} descriptor.

Reviewer: ```suggestion
The read-only **`overrideColors`** property of the {{domxref("CSSFontPaletteValuesRule")}} interface is a string containing a list of color index and color pair that are to be used instead. It is specified in the same format as the corresponding {{cssxref("@font-palette-values/override-colors", "override-colors")}} descriptor.
```

## Value

> A string containing a comma-separated list of pairs of color indices and colors.

Reviewer: ```suggestion
A string containing a comma-separated list of color index and color pair
```
